mmv1/compiler.rb

```diff
@@ -247,22 +232,11 @@
     product_api.set_variable(resources, 'objects')
   end
 
-  if File.exist?(provider_yaml_path)
-    product_api, provider_config, = \
-      Provider::Config.parse(provider_yaml_path, product_api, version)
-  end
-  # Load any dynamic overrides passed in with -r
-  if File.exist?(provider_override_path)
-    product_api, provider_config, = \
-      Provider::Config.parse(provider_override_path, product_api, version, override_dir)
-  end
-
-  Google::LOGGER.info "#{product_name}: Compiling provider config"
-  pp provider_config if ENV['COMPILER_DEBUG']
+  product_api&.validate
```

Comment:
`product_api&.validate` was inside `Provider::Config.parse` method and still needed.